### ISSUE TITLE
Handle custom HTTP requests via netty-socket.io

### ISSUE THREAD

**Issue by thangdao1992** [mrniko/netty-socketio#604]
I came across this configuration option:
setAllowCustomRequests Allow to service custom requests that differ from socket.io protocol, defaults to false. If true, add own handler which handle custom requests in order to avoid hang connections.

Has anyone used this before? Are there examples on how to add my own handler?

**Comment by thangdao1992** [comment]
@mrniko

**Comment by chemseddine201** [comment]
same problem for me any solution ?

**Comment by ZSMJ2014** [comment]
> same problem for me any solution ?

1.extend class SocketIOChannelInitializer,rewrite method addSocketioHandlers
2.extend class SimpleChannelInboundHandler,and impl your own handler
3.in the handler transfer FullRequest to HttpServletRequest,so you can deal with the request like HttpServletRequest;you need create an Adaptor class to do the transformation.

**Comment by chemseddine201** [comment]
> 1.extend class SocketIOChannelInitializer,rewrite method addSocketioHandlers
> 2.extend class SimpleChannelInboundHandler,and impl your own handler
> 3.in the handler transfer FullRequest to HttpServletRequest,so you can deal with the request like HttpServletRequest;you need create an Adaptor class to do the transformation.

yes but not work for me, i want to avoid socket and send http request from php server to netty-socketio server because i have a problem with handshake file too long time than php timeout, so i want to use an http request as a replacement for socket is it possible ?

**Comment by ZSMJ2014** [comment]
> > 1.extend class SocketIOChannelInitializer,rewrite method addSocketioHandlers
> > 2.extend class SimpleChannelInboundHandler,and impl your own handler
> > 3.in the handler transfer FullRequest to HttpServletRequest,so you can deal with the request like HttpServletRequest;you need create an Adaptor class to do the transformation.
> 
> yes but not work for me, i want to avoid socket and send http request from php server to netty-socketio server because i have a problem with handshake file too long time than php timeout, so i want to use an http request as a replacement for socket is it possible ?

use setTransports ? [https://github.com/mrniko/netty-socketio/wiki/Configuration-details](url)

**Comment by chemseddine201** [comment]
> > > 1.extend class SocketIOChannelInitializer,rewrite method addSocketioHandlers
> > > 2.extend class SimpleChannelInboundHandler,and impl your own handler
> > > 3.in the handler transfer FullRequest to HttpServletRequest,so you can deal with the request like HttpServletRequest;you need create an Adaptor class to do the transformation.
> > 
> > 
> > yes but not work for me, i want to avoid socket and send http request from php server to netty-socketio server because i have a problem with handshake file too long time than php timeout, so i want to use an http request as a replacement for socket is it possible ?
> 
> use setTransports ? [https://github.com/mrniko/netty-socketio/wiki/Configuration-details](url)

of course i'm using  `config.setTransports(Transport.WEBSOCKET, Transport.POLLING);`
the server worked fine but i want to improve it because it leak sometime and can't deliver the handshake data, so i want to ignore the handshake step and emit directly the event, thats why i want to use http request (POST by example)  to send data to the server. is it possible ?

**Comment by ZSMJ2014** [comment]
> > > > 1.extend class SocketIOChannelInitializer,rewrite method addSocketioHandlers
> > > > 2.extend class SimpleChannelInboundHandler,and impl your own handler
> > > > 3.in the handler transfer FullRequest to HttpServletRequest,so you can deal with the request like HttpServletRequest;you need create an Adaptor class to do the transformation.
> > > 
> > > 
> > > yes but not work for me, i want to avoid socket and send http request from php server to netty-socketio server because i have a problem with handshake file too long time than php timeout, so i want to use an http request as a replacement for socket is it possible ?
> > 
> > 
> > use setTransports ? [https://github.com/mrniko/netty-socketio/wiki/Configuration-details](url)
> 
> of course i'm using `config.setTransports(Transport.WEBSOCKET, Transport.POLLING);`
> the server worked fine but i want to improve it because it leak sometime and can't deliver the handshake data, so i want to ignore the handshake step and emit directly the event, thats why i want to use http request (POST by example) to send data to the server. is it possible ?

 try config.setTransports( Transport.POLLING); ?？
i think handshake step is necessary, can not skip it

**Comment by chemseddine201** [comment]
> > > > > 1.extend class SocketIOChannelInitializer,rewrite method addSocketioHandlers
> > > > > 2.extend class SimpleChannelInboundHandler,and impl your own handler
> > > > > 3.in the handler transfer FullRequest to HttpServletRequest,so you can deal with the request like HttpServletRequest;you need create an Adaptor class to do the transformation.
> > > > 
> > > > 
> > > > yes but not work for me, i want to avoid socket and send http request from php server to netty-socketio server because i have a problem with handshake file too long time than php timeout, so i want to use an http request as a replacement for socket is it possible ?
> > > 
> > > 
> > > use setTransports ? [https://github.com/mrniko/netty-socketio/wiki/Configuration-details](url)
> > 
> > 
> > of course i'm using `config.setTransports(Transport.WEBSOCKET, Transport.POLLING);`
> > the server worked fine but i want to improve it because it leak sometime and can't deliver the handshake data, so i want to ignore the handshake step and emit directly the event, thats why i want to use http request (POST by example) to send data to the server. is it possible ?
> 
> try config.setTransports( Transport.POLLING); ?？
> i think handshake step is necessary, can not skip it

thanks for responding to me, yes as i read on socket-io engine protocol the handshake is necessary but i want to avoid it by using an http request as you suggested in previous comments is possible ? also if you could share a code snippet to clarify to implementation to me.
